src/CodeViewer/Viewer.cpp

```diff
@@ -211,7 +211,8 @@ void Viewer::DrawLineNumbers(QPaintEvent* event) {
       const float intensity = std::clamp(
           static_cast<float>(num_samples_in_line) / static_cast<float>(num_samples_in_function),
           0.0f, 1.0f);
-      const auto scaled_intensity = static_cast<int>(std::sqrt(intensity) * 255);
+      const auto scaled_intensity =
+          std::clamp(static_cast<int>(std::sqrt(intensity) * 255), 0, 255);
```

Comment:
I don't understand the cause of this.

Is because static_cast <int> (0.0) is answering -1? Should we use round instead of static_cast?

Reply:
`static_cast<float>(num_samples_in_line) / static_cast<float>(num_samples_in_function)` is `NaN` when `num_samples_in_function` is 0. Then `intensity` gets `NaN` and `std::sqrt(intensity)` also returns `NaN`. The `static_cast<int>` invokes UB.

While thinking about it, maybe we should fix it differently.

Reply:
isn't the intensity clamped from 0.0 to 1.0?

Reply:
Doesn't matter. `clamp(Nan)` is still NaN.

Reply:
I pushed a revised version. PTAL!

Reply:
Yes, of course. thanks for explaining it.

Reply:
Sure. Thanks for asking!